woocommerce/woocommerce-ios
# [Mobile Payments] Pass gateway ID to existing IPP onboarding Tracks events

Closes: #10596 


## Why

Based on the issue description:

We currently do not pass the _payment gateway ID_ as an explicit property to some IPP onboarding events, and rely on the event identifier itself to know if is WCPay or others. Ideally, we want to pass this explicitly into the event property list, for example as we do in `_card_present_collect_payment_success`. This PR adds `plugin_slug` property to all 6 `card_present_onboarding_*` events.

## How

The 6 `card_present_onboarding_*` events were updated to include a new `plugin_slug` (gateway ID) property. `gatewayID: String?` extension was added to `CardPresentPaymentOnboardingState` for easier access in a few events. The changes are a bit scattered as the plugin needs to be DI'ed or inferred by looking at the code. I also deleted `CardPresentPaymentOnboardingState.enabled` case because it's not used anymore.

## Testing instructions


Hand-testing is optional, as it can be time-consuming to test all scenarios. There are also a lot of scenarios that could be hard to test like for a few error scenarios. I will make sure to ensure the event property `plugin_slug` is tracked for all 6 events.

- [x] @jaclync updates event registration in Tracks

---

- [x] `card_present_onboarding_step_skipped`
  - Find a store with the payment plugin all set up
  - In wp-admin > payment methods (`wp-admin/admin.php?page=wc-settings&tab=checkout`), disable `Cash on delivery` payment method and tap `Save changes`
  - In the app, switch to the store
  - Go to the Menu tab > Payments
  - Tap `Continue setup` --> should see a screen like in [this issue](https://github.com/woocommerce/woocommerce-ios/issues/7525))
  - Tap `Skip for now`
- [x] `card_present_onboarding_cta_tapped`, `card_present_onboarding_cta_failed`
  - Switch to a store with a payment plugin but not set up yet
  - Go to the Menu tab > Payments
  - Tap `Continue setup` 
  - Tap `Finish Setup in Store Admin` --> both events should be tracked with `plugin_slug`
- [x] `card_present_onboarding_learn_more_tapped`
  - Switch to a store with a payment plugin but not set up yet
  - Go to the Menu tab > Payments
  - Tap `Continue setup`
  - Tap `Learn more` --> the event should be tracked with `plugin_slug`
- [x] `card_present_onboarding_not_completed`
  - Switch to a store with a payment plugin but not set up yet
  - Go to the Menu tab > Payments
  - Tap `Continue setup`
  - Navigate back to the payments screen --> the event should be tracked with `plugin_slug`
- [x] `card_present_onboarding_completed`
  - Switch to a store with payments set up already
  - Go to the Menu tab > Payments --> the event should be tracked with `plugin_slug`

---
